Separate how to create k8s cluster guide (#200)
scalar-labs/scalar-kubernetes: docs/CreateAKSClusterForScalarDB.md

Note that you should configure resource limits based on your system's workload if the Envoy pod exceeds the above resource usage. Also, you should consider scaling out the worker node and the ScalarDB Server pod if all the ScalarDB Server pod exceeds the above resource usage and the latency is high (throughput is low) in your system.

### Create a node pool for ScalarDB Server pods

AKS creates one system node pool named **agentpool** that is preferred for system pods (used to keep AKS running) by default. It is recommended to create another node pool with **user** mode for ScalarDB Server pods and deploy ScalarDB Server pods on this additional node pool.

### Create a node pool for other application pods than ScalarDB Server pods

It is recommended to run only ScalarDB Server pods on the worker node (node pool) for ScalarDB Server. If you want to run other application pods on the same AKS, you should create other node pools for your application pods.

> ### Create a node group for monitoring components (kube-prometheus-stack and loki-stack)

Author: Ah, you're right. We should use `node pool` in AKS docs.
Sorry, it's my mistake...

> # Create an AKS cluster for ScalarDB Server

Reviewer: Very sorry for the late review.
For this, can it be converted for ScalarDB Cluster?
Or, we should keep it but mark it as deprecated?

Also, I feel the tile should be something like this since users don't create a cluster by following the document.
Thought?
```
A guideline for creating an AKS cluster for ScalarDB Server
```

Author: Thank you for your review!

> For this, can it be converted for ScalarDB Cluster?
> Or, we should keep it but mark it as deprecated?

I will create a new document for ScalarDB Cluster in another PR.
And, in that PR, I will make these ScalarDB Server documents deprecated.

> Also, I feel the tile should be something like this since users don't create a cluster by following the document.
> Thought?

Ah, I see. Indeed, these documents don't describe `how to create managed k8s cluster`. These documents describe only the requirements and recommendations.

I agree with your suggestion. I updated all documents in 2a63f33145871a88697194f734005123c3a551b3.

Note: We have to update link (document name) in other documents, but we are updating those related documents (i.e., it might cause conflict). So, I will update those links in another PR after related PRs will be merged.

> This document explains how to create an Azure Kubernetes Service (AKS) cluster for ScalarDB Server deployment. For details on how to deploy ScalarDB Server on an AKS cluster, see [Deploy ScalarDB Server on AKS](./ManualDeploymentGuideScalarDBServerOnAKS.md).

Reviewer: ```suggestion
This document explains the requirements and recommendations for creating an Azure Kubernetes Service (AKS) cluster for ScalarDB Server deployment. For details on how to deploy ScalarDB Server on an AKS cluster, see [Deploy ScalarDB Server on AKS](./ManualDeploymentGuideScalarDBServerOnAKS.md).
```
As mentioned above, users don't create a cluster by following the document, so it might be misleading to say `This document explains how to create an Azure Kubernetes Service`.

Author: I also updated this part in 2a63f33145871a88697194f734005123c3a551b3.

Note that you should configure resource limits based on your system's workload if the Envoy pod exceeds the above resource usage. In addition, you should consider scaling out the worker node and the ScalarDB Server pod if the ScalarDB Server pod exceeds the above resource usage and if latency is high (throughput is low) in your system.

### Create a node pool for ScalarDB Server pods

AKS creates one system node pool named **agentpool** that is preferred for system pods (used to keep AKS running) by default. We recommend creating another node pool with **user** mode for ScalarDB Server pods and deploying ScalarDB Server pods on this additional node pool.

### Create a node pool for other, non-ScalarDB Server application pods

We recommend running only ScalarDB Server pods on the worker node (node pool) for ScalarDB Server. If you want to run other application pods on the same AKS cluster, you should create other node pools for your application pods.

### Create a node pool for monitoring components (kube-prometheus-stack and loki-stack)

> We recommend running only ScalarDB Server pods on the worker node (node pool) for ScalarDB Server. If you want to run monitoring pods (e.g., Prometheus, Grafana, Loki, etc.) by using [kube-prometheus-stack](./K8sMonitorGuide.md) and [loki-stack](./K8sLogCollectionGuide.md) on the same AKS cluster, you should create other node groups for monitoring pods.

Reviewer 2: ```suggestion
We recommend running only ScalarDB Server pods on the worker node (node pool) for ScalarDB Server. If you want to run monitoring pods (e.g., Prometheus, Grafana, Loki, etc.) by using [kube-prometheus-stack](./K8sMonitorGuide.md) and [loki-stack](./K8sLogCollectionGuide.md) on the same AKS cluster, you should create other node pools for monitoring pods.
```

Author: Ah, sorry. I overlooked this point in my previous updates.
Thank you for pointing it out!

* [Use kubenet networking with your own IP address ranges in Azure Kubernetes Service (AKS)](https://learn.microsoft.com/en-us/azure/aks/configure-kubenet)
* [Configure Azure CNI networking in Azure Kubernetes Service (AKS)](https://learn.microsoft.com/en-us/azure/aks/configure-azure-cni)

### Create the AKS cluster on a private network

You should create the AKS cluster on a private network (private subnet in a VNet) since ScalarDB Server does not provide any services to users directly via internet access. We recommend accessing ScalarDB Server via a private network from your applications.

### Create the AKS cluster by using Azure CNI, if necessary

The AKS default networking plug-in is [kubenet](https://learn.microsoft.com/en-us/azure/aks/configure-kubenet). If your requirement does not match kubenet, you should use [Azure Container Networking Interface (CNI)](https://learn.microsoft.com/en-us/azure/aks/configure-azure-cni).

> For example, if you want to deploy multiple ScalarDB Server environments on one AKS cluster (e.g., deploy a multi-tenant ScalarDB Server) and you want to control the connection between each tenant by using [Kubernetes NetworkPolicies](https://kubernetes.io/docs/concepts/services-networking/network-policies/), kubenet supports only Calico Network Policy, which the [Azure support team does not support](https://learn.microsoft.com/en-us/azure/aks/use-network-policies#differences-between-azure-network-policy-manager-and-calico-network-policy-and-their-capabilities). Please note that Calico Network Policy is supported only by the Calico community or through additional paid support.

Reviewer 2: ```suggestion
For example, if you want to deploy multiple ScalarDB Server environments on one AKS cluster (e.g., deploy a multi-tenant ScalarDB Server) and you want to control the connection between each tenant by using [Kubernetes NetworkPolicies](https://kubernetes.io/docs/concepts/services-networking/network-policies/), kubenet supports only the Calico Network Policy, which the [Azure support team does not support](https://learn.microsoft.com/en-us/azure/aks/use-network-policies#differences-between-azure-network-policy-manager-and-calico-network-policy-and-their-capabilities). Please note that the Calico Network Policy is supported only by the Calico community or through additional paid support.
```
